src/main/java/org/spongepowered/api/item/inventory/transaction/InventoryTransactionResult.java

```diff
@@ -74,50 +76,84 @@ public static InventoryTransactionResult failNoTransactions() {
         return InventoryTransactionResult.builder().type(Type.ERROR).build();
     }
 
+    /**
+     * The type of InventoryTransactionResult.
+     */
     public enum Type {
 
-        /**
-         * The actual result of the operation is undefined, this probably
-         * indicates that something went wrong with the operation that the
-         * inventory couldn't handle or didn't expect. The state of the
-         * inventory is undefined.
-         */
-        UNDEFINED,
-
         /**
          * The inventory operation succeeded.
+         *
+         * <p>The state of the inventory may have changed.</p>
          */
         SUCCESS,
 
         /**
          * The inventory operation failed for an <em>expected</em> reason (such
-         * as the inventory being full or not accepting items of a supplied
-         * type. The condition of the inventory is unchanged.
+         * as the inventory being full, not accepting items of a supplied type
+         * or a third party fully or partially canceling the transactions.
+         *
+         * <p>The state of the inventory may have changed.</p>
          */
         FAILURE,
 
         /**
          * The inventory operation failed because an <em>unexpected</em>
-         * condition occurred. The state of the inventory is undefined.
+         * condition occurred.
+         *
+         * <p>The state of the inventory is undefined.</p>
          */
-        ERROR,
+        ERROR
 
-        /**
-         * An operation was cancelled by a third party (eg. an inventory event
-         * was cancelled). The condition of the inventory is unchanged.
-         */
-        CANCELLED
-        
     }
 
-    final Type type;
+    private final List<SlotTransaction> slotTransactions;
     private final List<ItemStackSnapshot> rejected;
-    private final List<ItemStackSnapshot> replaced;
+
+    final Type type;
 
     InventoryTransactionResult(Builder builder) {
```

Comment:
Maybe move the `InventoryTransactionResult` impl to SpongeCommon?

Reply:
Yes please ^